src/cargo/core/compiler/layout.rs

```diff
@@ -167,6 +174,16 @@ impl Layout {
         let root = root.into_path_unlocked();
         let dest = dest.into_path_unlocked();
 
+        // Compute the sysroot path for the build-std feature.
+        let build_std = ws.config().cli_unstable().build_std.as_ref();
+        let (sysroot, sysroot_libdir) = if let Some(tp) = build_std.and(triple_path) {
+            let sysroot = dest.join(".sysroot");
+            let sysroot_libdir = sysroot.join("lib").join("rustlib").join(tp).join("lib");
```

Comment:
So technically this isn't going to work on all compilers digging in a bit. The [sysroot finding function in rustc](https://github.com/rust-lang/rust/blob/6ef275e6c3cb1384ec78128eceeb4963ff788dca/src/librustc/session/filesearch.rs#L143-L169) takes into account `CFG_LIBDIR_RELATIVE` which is set by taking the difference between `--libdir` and `--prefix` when building rustc itself.

Now that's pretty esoteric though and only really relevant for Linux distributions, but it may be worth keeping in mind that this may run afoul of that at some point.

Reply:
Oh…  I'll try to investigate.  There doesn't seem to be a way to discover the actual path.